cmd/virtual-kubelet/internal/commands/root/root.go

```diff
@@ -207,11 +206,15 @@ func runRootCommand(ctx context.Context, s *provider.Store, c Opts) error {
 	}()
 
 	if c.StartupTimeout > 0 {
-		// If there is a startup timeout, it does two things:
-		// 1. It causes the VK to shutdown if we haven't gotten into an operational state in a time period
-		// 2. It prevents node advertisement from happening until we're in an operational state
-		err = waitFor(ctx, c.StartupTimeout, pc.Ready())
-		if err != nil {
+		ctx, cancel := context.WithTimeout(ctx, c.StartupTimeout)
+		log.G(ctx).Info("Waiting for pod controller / VK to be ready")
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-pc.Ready():
+		}
+		cancel()
```

Comment:
Shouldn't this be deferred?

Reply:
Isn't this the context just for startup, and not the whole lifecycle?

Reply:
Right, though it was missing a cancel in the `.Done()` branch.

Reply:
if ctx.Done() returns, doesn't that mean it's cancelled / done?

Reply:
Generally yes, but not if the parent is the thing that was cancelled.

Reply:
And yes, that through me too :)
Linter picked it up.